app/src/main/java/com/example/superimagesviewer/viewmodel/MosaicViewModel.kt

```diff
@@ -1,16 +1,46 @@
 package com.example.superimagesviewer.viewmodel
 
 import android.app.Application
-import androidx.lifecycle.AndroidViewModel
-import androidx.lifecycle.LiveData
 import android.graphics.drawable.Drawable
+import android.util.Log
+import androidx.lifecycle.*
+import com.example.superimagesviewer.Settings
 import com.example.superimagesviewer.repository.MosaicRepository
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.launch
+import kotlinx.coroutines.withContext
 
 class MosaicViewModel(application: Application) : AndroidViewModel(application) {
-    val mosaicsList: LiveData<List<Drawable>>
-    private val mosaicRepository: MosaicRepository = MosaicRepository(application)
+
+    private val TAG = MosaicViewModel::class.simpleName
+
+    public val drawablesList: LiveData<List<Drawable>>
+    public val userName: LiveData<String>
+    private val mosaicRepository: MosaicRepository = MosaicRepository()
 
     init {
-        mosaicsList = mosaicRepository.getMosaicsList()
+        viewModelScope.launch {
+            withContext(Dispatchers.IO) {
+                if (mosaicRepository.isNetworkConnected(application)) {
+                    Log.d(TAG, "Successful internet connection")
```

Comment:
I think this log is not required anymore

Reply:
Done